### DISCUSSION TITLE
nnapi : Model execution is not supported in this build. Help!!!

### DISCUSSION THREAD

**Discussion by deahhh** [microsoft/onnxruntime#20266] [API Q&A]
I have built libonnxruntime.so.1.17.1 with command:
```./build.sh --config MinSizeRel --minimal_build extended --use_nnapi --parallel --compile_no_warning_as_error  --disable_ml_ops --disable_exceptions --build_shared_lib --skip_tests --include_ops_by_config ./deploy/onnx_models/required_operators.config --skip_submodule_sync --cmake_extra_defines FETCHCONTENT_TRY_FIND_PACKAGE_MODE=NEVER```
on linux amd64.

File `required_operators.config` is generated with command `python -m onnxruntime.tools.convert_onnx_models_to_ort ./deploy/onnx_models  --target_platform amd64 --optimization_style Runtime`.

model.onnx is generated from pytorch:
```
class Resize(nn.Module):
    def __init__(self,) -> None:
        super().__init__()

    def forward(self, x : torch.Tensor, shape: torch.Tensor):
        x = x.unsqueeze(0).unsqueeze(0)
        x = nn.functional.interpolate(x, size=(shape[0].item(), shape[1].item())).type(torch.float32)
        x = (x-x.mean())#/(x.std()+1e-6)
        return x
```
![resize with_runtime_opt ort](https://github.com/microsoft/onnxruntime/assets/23054573/18327345-0766-46d0-bc16-baab035596d0)

Load the ort model with cpp, build and run:
```
uint32_t nnapi_flags = 
nnapi_flags |= NNAPI_FLAG_USE_NCHW|NNAPI_FLAG_CPU_DISABLED;//NNAPI_FLAG_CPU_ONLY;
Ort::ThrowOnError(OrtSessionOptionsAppendExecutionProvider_Nnapi(*session_options_, nnapi_flags));
```
I got error message as follow:
```
[E:onnxruntime:, sequential_executor.cc:514 ExecuteKernel] Non-zero status code returned while running Nnapi_18077046291140798866_0 node. Name:'NnapiExecutionProvider_Nnapi_18077046291140798866_0_0' Status Message: Model execution is not supported in this build.
terminate called after throwing an instance of 'Ort::Exception'
  what():  Non-zero status code returned while running Nnapi_18077046291140798866_0 node. Name:'NnapiExecutionProvider_Nnapi_18077046291140798866_0_0' Status Message: Model execution is not supported in this build.
```

Comment `x = (x-x.mean())` will executes it correctly.

Help! I have followed the [instructions](https://onnxruntime.ai/docs/execution-providers/NNAPI-ExecutionProvider.html) to the letter. And the ops used here are all listed in the page. What's WRONG with my code!

**Comment by Fatih-Kiralioglu** [comment]
Hi [deahhh](https://github.com/deahhh), have you tried this test in an android device or linux os target? I am getting the same error with Mac OS target. I suspect  nnapi execution provider is supported in only Android.
